Wrappers/Python/ccpi/optimisation/functions/MixedL21Norm.py

```diff
@@ -19,105 +19,164 @@
 from __future__ import absolute_import
 from __future__ import division
 from __future__ import print_function
-from __future__ import unicode_literals
 
-from ccpi.optimisation.functions import Function, ScaledFunction
+
+from ccpi.optimisation.functions import Function
 from ccpi.framework import BlockDataContainer
 import numpy as np
 
-import functools
 
 class MixedL21Norm(Function):
     
     
-    '''
-        f(x) = ||x||_{2,1} = \sum |x|_{2}                   
-    '''      
+    """ MixedL21Norm function: :math:`F(x) = ||x||_{2,1} = \sum |x|_{2} = \sum \sqrt{ (x^{1})^{2} + (x^{2})^{2} + \dots}`                  
+    
+        where x is a BlockDataContainer, i.e., :math:`x=(x^{1}, x^{2}, \dots)`
+    
+    """      
     
     def __init__(self, **kwargs):
 
-        super(MixedL21Norm, self).__init__()                      
+        super(MixedL21Norm, self).__init__()  
+        self.b = kwargs.get('b', None)  
+
+        # This is to handle tensor for Total Generalised Variation                    
```

Comment:
When true , we use it for Total Generalied Variation, see [here](https://github.com/vais-ral/CCPi-Framework/blob/merge_master_into_Functions/Wrappers/Python/ccpi/framework/BlockGeometry.py#L62)